### DISCUSSION TITLE
$caller

### DISCUSSION THREAD

**Discussion by ajmalton** [DivinumOfficium/divinum-officium#2567] [Developers corner]
Anyone remember what $caller is/was for? Something something All Souls day... but it _looks_ obsolete now. Unless I’ve missed something.

**Comment by gregordick** [comment]
It looks like it's used to distinguish between All Souls' day and the office of the dead said for other reasons.  The former ends after None, but the latter is said whenever the user requests it: https://github.com/DivinumOfficium/divinum-officium/blob/f35b45583170ab553848a24f0be338c5bad30e5f/web/cgi-bin/horas/horascommon.pl#L487-L495

This results in these two behaving differently from one another:

https://divinumofficium.com/cgi-bin/horas/Pofficium.pl?date1=11-2-2021&command=prayVespera&version=Divino%20Afflatu&testmode=regular&lang2=English&votive=&caller=1
https://divinumofficium.com/cgi-bin/horas/Pofficium.pl?date1=11-2-2021&command=prayVespera&version=Divino%20Afflatu&testmode=regular&lang2=English&votive=

And then there's this: https://github.com/DivinumOfficium/divinum-officium/blob/f35b45583170ab553848a24f0be338c5bad30e5f/web/cgi-bin/horas/horascommon.pl#L371

The intention isn't obvious to me, but it feels like a workaround for some interaction between 11-02t (which is the day within the octave of All Saints... or would be if that file existed...) and the explicitly-requested office of the dead.

**Reply by ajmalton** [reply]
Yes, it looks like that was the intent.  However it seems to be broken, and unless I've missed it there is no UI that allows the feature to be used.

**Reply by gregordick** [reply]
It looks like it could in principle be triggered via the UI, using the links generated using the `%...%` syntax, if we populated the `dirge` lines in the transfer tables.  But we don't.  I think it's fine to purge if this would make life easier.
